### DISCUSSION TITLE
Added a large file to local and now I can not push

### DISCUSSION THREAD

**Discussion by dapomGH** [community/community#145940] [New to GitHub]
3 days ago I stupidly added a 215MB file to my project and included it into the VCS.

Unfortunately I made numerous other changes and tried deleting the file/directory a number of times.  I also updated after deleting and the file/directory reappeared.

It was pushed to GitHub. Where I saw it in its original size.

I have since tried to delete it and push I keep getting push failures saying it is larger than the limit of 100MB.

I do not see it in my local repo but I guess it is trying to push a prior commit where I added it. (how it ever got to GitHub I do not understand but …).

I just want to get it removed from everyplace and go on with my project.

Any suggestions would be appreciated.

I did not want to clutter up the above with all my attempts to correct this situation. I tried increasing the buffer and installing LFS (and migrating the existing file to LFS)
and at least that got rid of the RPC failed the remote hung up unexpectantly.

I also tried to go to githup and edit the file down to a few bytes but to no avail.

So I am stuck.

I saw a suggestion to clone the project but I would only want to do that as a last resort (would that even work?).

Finally, As I mentioned I increase the post buffer - how do I set it back to default size? (I assume I should???).

Thanks

PS I am working through GoLand on a windows 11 machine

**Comment by nexoscreator** [comment]
Use GitHub Desktop For Uplaoding Larger Files

[Download GitHub](https://desktop.github.com/download/)

Also you need stable Internet Connection

**Comment by stack-dhruv** [comment]
You can explore: https://rtyley.github.io/bfg-repo-cleaner/

--
Another solution is: If you’ve committed the file but haven’t yet pushed it, you can remove it from your Git history and ensure it doesn’t appear in the remote repository when you push. 

### Steps to Remove the File Before Pushing

#### 1. **Remove the File from Git's Tracking**
Use `git rm` to remove the file from Git while keeping it in your local filesystem.

```bash
git rm --cached <file-path>
```
- `--cached`: Removes the file only from the staging area (not from your local filesystem).

#### 2. **Add the Removal to a New Commit**
Create a new commit to register the file’s removal in Git.

```bash
git commit -m "Remove large file from tracking"
```

#### 3. **Push the Changes**
Now you can push your branch to the remote repository.

```bash
git push origin <branch-name>
```

This will successfully push your changes without including the large file in the remote history.

- This method works <mark>**only if the file hasn’t been pushed yet**.</mark> If it’s already in the remote repository’s history, you’ll need to rewrite the history as outlined earlier.
- If the file was accidentally included in `.gitignore`, add it to `.gitignore` to prevent it from being tracked again.

```bash
echo "<file-path>" >> .gitignore
git add .gitignore
git commit -m "Add file to .gitignore"
git push origin <branch-name>
```
